test/new-e2e/tests/containers/docker_test.go

```diff
@@ -0,0 +1,98 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package containers
+
+import (
+	"context"
+	"fmt"
+	fakeintake "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/infra"
+	dockervm "github.com/DataDog/test-infra-definitions/scenarios/aws/dockerVM"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/stretchr/testify/suite"
+	"os"
+	"testing"
+)
+
+type DockerSuite struct {
+	baseSuite
+}
+
+func TestDockerSuite(t *testing.T) {
+	suite.Run(t, &DockerSuite{})
+}
+
+func (suite *DockerSuite) SetupSuite() {
+	ctx := context.Background()
+
+	stackConfig := runner.ConfigMap{
+		"ddagent:deploy":     auto.ConfigValue{Value: "true"},
+		"ddagent:fakeintake": auto.ConfigValue{Value: "true"},
+	}
+
+	_, stackOutput, err := infra.GetStackManager().GetStack(ctx, "docker-stack", stackConfig, dockervm.Run, false)
+	suite.Require().NoError(err)
+
+	fakeintakeHost := stackOutput.Outputs["fakeintake-host"].Value.(string)
+	suite.Fakeintake = fakeintake.NewClient(fmt.Sprintf("http://%s", fakeintakeHost))
+
+	vmConnection := stackOutput.Outputs["vm-connection"].Value.(map[string]interface{})
+	suite.clusterName = fmt.Sprintf("%s-%v", os.Getenv("USER"), vmConnection["host"])
+
+	suite.baseSuite.SetupSuite()
+}
+
+func (suite *DockerSuite) TestDSDWithUDS() {
+	suite.testMetric(&testMetricArgs{
+		Filter: testMetricFilterArgs{
+			Name: "custom.metric",
+			Tags: []string{
+				`container_name:metric-sender-uds`,
+			},
+		},
+		Expect: testMetricExpectArgs{
+			Tags: &[]string{
+				`^container_name:metric-sender-uds$`,
+				`^short_image`,
+				`^image_tag`,
+				`^docker_image`,
+				`^git.commit.sha`,
+				`^git.repository_url`,
+				`^series`,
+				`^image_name`,
+				`^image_id`,
+				`^container_id`,
```

Comment:
Could you sort the tags by alphabetical order ?
It might be helpful when looking for a specific tag when debugging.

Also, as the image is hard-coded on `test-infra-definitions` side, we must probably be able to assert not only tags keys but also some tags values or a least a prefix. Like what we do in other tests like: https://github.com/DataDog/datadog-agent/blob/f80d187f533f5c7dee5c184e95fe632dfa720f40/test/new-e2e/tests/containers/k8s_test.go#L608-L627

The idea is to assert as much things as possible.
For the `container_id` tags, we cannot predict its value in advance. That’s why we can only assert that this tag is present.
But for many others like `short_image`, `image_tag`, etc., I think we can assert the tag value.